hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/Storage.java

```diff
@@ -52,6 +53,19 @@
 
     Collection<R> values();
 
+    /**
+     * Fail fast may not be the desired behaviour always. For example if you are caching an iterator as in
+     * {@link AbstractEvictableRecordStore#expirationIterator} since you know that in next rounds you will
+     * eventually visit all entries.
+     *
+     * Note that returned iterator is not thread-safe !!!
+     *
+     * @param failFast {@code true} if the returned iterator should fail-fast in case of concurrent modification
+     *                            during iteration otherwise set {@code false}
+     * @return new iterator instance
+     */
+    Iterator<R> iterator(boolean failFast);
```

Comment:
since it is explained in java doc, i don't expect big pain + i like shorter name here